crates/editor/src/editor.rs

```diff
@@ -3328,6 +3328,33 @@ impl Editor {
         Subscription::join(other_subscription, this_subscription)
     }
 
+    fn unfold_buffers_with_selections(&mut self, cx: &mut Context<Self>) {
+        if self.buffer().read(cx).is_singleton() {
+            return;
+        }
+
+        let snapshot = self.buffer.read(cx).snapshot(cx);
+        let buffers_to_unfold: Vec<BufferId> = self
+            .selections
+            .disjoint_anchor_ranges()
+            .flat_map(|range| {
+                // Check all buffers that this selection spans
+                let start_offset = range.start.to_offset(&snapshot);
```

Comment:
I think this conversion might be redundant, since `buffer_ids_for_range` can also take `Range<Anchor>`, right?